### PR TITLE
Introduced InteropContextInspector 

### PR DESCRIPTION
**Pull request by dgquintas**
In order to be able to peek into server contexts during interop testing.

This is just the skeleton, as the motivating usecase (compression) lives in #2135.

### CODE REVIEW

File: test/cpp/interop/server_helper.h [hunk 1]
```diff
@@ -43,6 +44,16 @@ namespace testing {
 
 std::shared_ptr<ServerCredentials> CreateInteropServerCredentials();
 
+class InteropContextInspector {
+ public:
+  InteropContextInspector (const ::grpc::ServerContext& context);
```
Comment (ctiller) [inline test/cpp/interop/server_helper.h]: remove space/run clang-format

Comment (dgquintas) [inline test/cpp/interop/server_helper.h]: Done. Note that msan complains about context_ not being used. It'll be used alright in a following PR.

Comment (ctiller) [inline test/cpp/interop/server_helper.h]: I'll ensure the merge window is short :)

Comment (dgquintas) [inline test/cpp/interop/server_helper.h]: Actually, lemme introduce a placeholder method to be removed once an actually useful one takes its place to appease the gods of msan.

**Comment by dgquintas** [comment]
PTAL. The change to server context (for the IsCancelled() const-ness) could (and should) go into a separate PR, but well...

**Pull request merged**